airflow/jobs.py

```diff
@@ -110,6 +110,7 @@ def __init__(
         self.latest_heartbeat = timezone.utcnow()
         self.heartrate = heartrate
         self.unixname = getpass.getuser()
+        self.max_tis_per_query = conf.getint('scheduler', 'max_tis_per_query')
```

Comment:
Maybe set a default value in this for the configuration templates?

Reply:
# This changes the batch size of queries in the scheduling main loop.                     
# This depends on query length limits and how long you are willing to hold locks.         
# 0 for no limit                                                                          
max_tis_per_query = 0

Reply:
My PR uses an existing config parameter that was used to implement batching, but for only -one- of the problematic queries. I've simply refactored and extended that approach.

So, the config parameter is actually already in the template, but the default is 0.

If we leave it at 0, Airflow's current behavior will be preserved. However, it might make sense to change the default to 512, which I've found to work well in practice.

WDYT?

Reply:
Yes, please add it to the config with a description what the configuration parameter does. I think it would be good to set it to 512, since it will improve performance.

Reply:
I've set it to 512 as a default, and added a more detailed comment